v4/export/ir.go

```diff
@@ -27,9 +27,9 @@ type SQLRowIter interface {
 	Decode(RowReceiver) error
 	Next()
 	Error() error
-	HasNext() bool
-	HasNextSQLRowIter() bool
-	NextSQLRowIter() SQLRowIter
+	HasNext(currentStatementSize, currentFileSize uint64) bool
+	HasNextSQLRowIter(currentFileSize uint64) bool
+	AdjustFileRowIterSize(currentStatementSize, currentFileSize uint64)
```

Comment:
why must these passed as arguments into the iterator?

Reply:
Because currently the written size is saved in `writer` but the `FilesizeLimit` and `StatementsizeLimit` is saved in file row iterator

Reply:
gotta say this solution doesn't look good to me. the `SQLRowIter` should only care about pumping out `*sql.Row`s, it should be the CSVWriter/SQLWriter's responsibility to know when to add the statement separator and when to create a new file.

Reply:
I see. I will take thought on how to refine this.

Reply:
revised, PTAL again